mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/IdentifierBridge.java

```diff
@@ -50,6 +51,18 @@ default boolean isCompatibleWith(IdentifierBridge<?> other) {
 		return equals( other );
 	}
 
+	/**
+	 * Parse an input String to the raw index field value.
+	 *
+	 * @param value The value to parse.
+	 * @return The raw index field value.
+	 * @throws RuntimeException If the value cannot be parsed to the raw index field value.
+	 */
+	@Incubating
+	default String parseToString(String value) {
+		return value;
+	}
```

Comment:
I couldn't use the  `parse(string)` method name here as it clashes with the parse method from the bridge interface, which has a different return type. Otherwise this last commit is about allowing `PARSE` on match-id predicates.

Reply:
>  You mean there's a clash for classes implementing both IdentifierBridge and ValueBridge?

yes, that 😃 

```java
public interface ValueBridge<V, F> {
    // @param <F> The type of raw index field values, on the index side of the bridge.
    default F parse(String value)
}
```
so parse returns the document value... (which is string for the id)

> Regardless... It's weird that this method returns a String rather than a I, no? I'd have expected Hibernate Search to apply toDocumentIdentifier automatically on the resulting "parsed" value (of type I), if necessary. Then your parseIdentifierLiteral method (or whatever you name it) could possibly be used for something else in the future.

mmm so it'll be something like:

```java
	@Override
	public String toDocumentValue(String value, ToDocumentValueConvertContext context) {
		return bridge.toDocumentIdentifier( bridge.parseIdentifierLiteral( value ), context );
	}
```

Reply:
> mmm so it'll be something like:
> 
> ```java
> 	@Override
> 	public String toDocumentValue(String value, ToDocumentValueConvertContext context) {
> 		return bridge.toDocumentIdentifier( bridge.parseIdentifierLiteral( value ), context );
> 	}
> ```

Either that or you'd handle it on the backend side, with something like `dslConverter.toDocumentValue(parserConverter.toDocumentValue(string, context), context)`.

I don't know what feels more consistent with what you do for fields :thinking: